### PR TITLE
[api] [error-handling] When checking a response header, ensure that t…

### PR DESCRIPTION
**Pull request by fajardoleo**
…he response is an array as it can also be a WP_Error instance.

### CODE REVIEW

File: includes/sdk/FreemiusWordPress.php [hunk 1]
```diff
@@ -354,8 +354,11 @@ static function RemoteRequest( $pUrl, $pWPRemoteArgs ) {
             $response = wp_remote_request( $pUrl, $pWPRemoteArgs );
 
             if (
-                empty( $response['headers'] ) ||
-                empty( $response['headers']['x-api-server'] )
+                is_array( $response ) &&
```
Comment (swashata) [inline includes/sdk/FreemiusWordPress.php:357]: @fajardoleo The `wp_remote_request` is guaranteed to return either an `array` or `WP_Error`. This is what we are doing in this method (+ the replica method in `class-fs-api.php`) too.

So I propose to just enhance the `@return` of this function and also `FS_Api#remote_request` to `@return array|WP_Error`.

Now about this `is_array` check, I was thinking if `!is_wp_error( $response )` is more semantic here. Please take a 2nd look.

File: includes/class-fs-api.php [hunk 2]
```diff
@@ -386,8 +386,11 @@ static function remote_request( $url, $remote_args ) {
             $response = wp_remote_request( $url, $remote_args );
 
             if (
-                empty( $response['headers'] ) ||
-                empty( $response['headers']['x-api-server'] )
+                is_array( $response ) &&
```
Comment (swashata) [inline includes/class-fs-api.php:389]: @fajardoleo Please see below.

**Review by swashata — COMMENTED** [review]
@fajardoleo Please see my small suggestion.

File: includes/sdk/FreemiusWordPress.php (hunk 1 above)
Comment (fajardoleo) [reply includes/sdk/FreemiusWordPress.php:357]: @swashata Thanks! The PHP documentation comment has been updated.

**Review by swashata — APPROVED** [review]
Thanks @fajardoleo LGTM.

**Pull request merged**